src/features/campaigns/campaignSlice.js

```diff
@@ -78,10 +79,33 @@ const campaignSlice = createSlice({
     },
     deleteNPC: (state, action) => {
       state.npcs = state.npcs.filter(npc => npc.id !== action.payload.id);
+    },
+    setCurrentObjectiveIndex: (state, action) => {
+      state.currentObjectiveIndex = action.payload
+    },
+    nextObjective: (state) => {
+      const currentQuestData = state.quests.find(quest => quest.title === state.currentQuest)
+      if (!currentQuestData) {
+        console.error("Current quest data not found in state.quests");
+        return;
+      }
+      if (typeof state.currentObjectiveIndex !== 'number') {
+        state.currentObjectiveIndex = 0;
+      } else if (state.currentObjectiveIndex < currentQuestData.objectives.length - 1) {
+        state.currentObjectiveIndex += 1;
+      }
+      console.log(state.currentObjectiveIndex)
+  },
+    previousObjective: (state) => {
+      const currentQuestData = state.quests.find(quest => quest.title === state.currentQuest);
+      if (currentQuestData && state.currentObjectiveIndex > 0) {
+        state.currentObjectiveIndex -= 1;
+      }
+      console.log(state.currentObjectiveIndex)
     }
   }
 })
 
-export const { currentQuestUpdated, addNote, deleteNote, editNote, addCharNote, editCharNote, setActiveNotes, setCurrentCampaign, addNPC, editNPC, deleteNPC } = campaignSlice.actions
+export const { currentQuestUpdated, addNote, deleteNote, editNote, addCharNote, editCharNote, setActiveNotes, setCurrentCampaign, addNPC, editNPC, deleteNPC, setCurrentObjectiveIndex, nextObjective, previousObjective } = campaignSlice.actions
 
 export default campaignSlice.reducer
```

Comment:
Add a state value and action to manage questModalVisible, then if we decide other screens should be able to manage the modal it'll be easy to integrate